### PR TITLE
test(hx-div): surf2002

### PR DESCRIPTION
**Pull request by lalithkarikelli**
## Description

Unit Test Case `<hx-div>`

image

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF-2002

## Before you request a review for this PR:

- [x] Did you add component tests for any new code?
- [x] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [x] Did you include a screenshot of the component tests?
- [x] If you changed/added functionality, did you update the demo page and documentation?
- [x] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/elements/hx-div/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    describe('test <hx-div> getter and setter methods', () => {
+        it('should be able to scroll in the vertical direction', async () => {
+            const component = /** @type {HXDivElement} */ await fixture(template);
+            component.scroll ='vertical';
+            const attr = component.hasAttribute('scroll');
+            const scrollDirection = component.getAttribute('scroll');
+
+            expect(attr).to.be.true;
+            expect(component.scroll).to.be.equal(scrollDirection);
```
Comment (100stacks) [inline src/elements/hx-div/index.spec.js:78]: Please switch the two variables `component.scroll` and `scrollDirection`:

```js
expect(scrollDirection).to.equal(component.scroll);
```

Comment (lalithkarikelli) [reply src/elements/hx-div/index.spec.js:78]: Changes have been done, please review.

File: src/elements/hx-div/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        });
+
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/elements/hx-div/index.spec.js:80]: Changes have been done, please review.

File: src/elements/hx-div/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                expect(len).to.be.equal(1);
+            });
+
+            it('should have an unnamed <slot>', async () => {
+                const component = /** @type { HXDivElement } */ await fixture(template);
+                const name = component.slot;
+
+                expect(name).to.be.equal('');
+            });
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/elements/hx-div/index.spec.js:65]: Changes have been done, please review.

File: src/elements/hx-div/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+            expect(prop).to.be.false;
+        });
+
+        it(`the rendered Light DOM should NOT equal simple template ${template}`, async () => {
+            const component = /** @type {HXDivElement} */ await fixture(template);
+
+            expect(component).lightDom.to.not.equal(template);
+        });
+
```
Comment (100stacks) [inline src/elements/hx-div/index.spec.js:32]: Please remove extra blank line.

File: src/elements/hx-div/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                const name = component.slot;
+
+                if (name !== null) {
+                    expect(name).to.be.equal('');
+                } else {
+                    expect(name).to.be.null;  // IE11, Legacy Edge, and older browsers
+                }
+            });
+        });
+
```
Comment (100stacks) [inline src/elements/hx-div/index.spec.js:71]: Please remove extra blank line.

File: src/elements/hx-div/index.spec.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+            const component = /** @type {HXDivElement} */ await fixture(template);
+            component.scroll = 'vertical';
+            const attr = component.hasAttribute('scroll');
+            const scrollDirection = component.getAttribute('scroll');
+
+            expect(attr).to.be.true;
+            expect(scrollDirection).to.be.equal(component.scroll);
+        });
+    });
+
```
Comment (100stacks) [inline src/elements/hx-div/index.spec.js:85]: Please remove extra blank line.

Comment (lalithkarikelli) [reply src/elements/hx-div/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-div/index.spec.js (hunk 5 above)
Comment (lalithkarikelli) [reply src/elements/hx-div/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-div/index.spec.js (hunk 4 above)
Comment (lalithkarikelli) [reply src/elements/hx-div/index.spec.js]: Changes have been done, please review.

**Pull request merged**